core/src/rules/optimising_line_formatter/mod.rs

```diff
@@ -1007,7 +1007,7 @@ impl<'this> InternalOptimisingLineFormatter<'this, '_> {
                 _ => 3,
             },
             RawDecision::Continue if decision.line_length > self.settings.max_line_length => {
-                2u64.pow(20)
+                2u64.pow(20) + (decision.line_length - self.settings.max_line_length) as u64 * 3
```

Comment:
Some context in a comment would be useful. Also, extract the '3' into a constant.